### PR TITLE
check for null key and value before cache.put

### PR DESCRIPTION
**Pull request by acha-bill**
# Description

Added null check before cache put. Fixes 
```
01/26 20:54:46.064 [Milestone Solidifier] ERROR MilestoneSolidifier:465 - Error while solidifying milestone #1071451
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011) ~[na:1.8.0_181]
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006) ~[na:1.8.0_181]
	at com.iota.iri.cache.impl.CacheImpl.put(CacheImpl.java:107) ~[iri-PR-e1526d2-e8e23c4c.jar:PR-e1526d2-e8e23c4c]
	at com.iota.iri.cache.impl.CacheImpl.get(CacheImpl.java:78) ~[iri-PR-e1526d2-e8e23c4c.jar:PR-e1526d2-e8e23c4c]
	at com.iota.iri.controllers.TransactionViewModel.fromHash(TransactionViewModel.java:158) ~[iri-PR-e1526d2-e8e23c4c.jar:PR-e1526d2-e8e23c4c]
	at com.iota.iri.TransactionValidator.checkSolidity(TransactionValidator.java:264) ~[iri-PR-e1526d2-e8e23c4c.jar:PR-e1526d2-e8e23c4c]
	at com.iota.iri.service.milestone.impl.MilestoneSolidifierImpl.isSolid(MilestoneSolidifierImpl.java:344) [iri-PR-e1526d2-e8e23c4c.jar:PR-e1526d2-e8e23c4c]
```

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

# How Has This Been Tested?

- existing unit tests pass

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/main/java/com/iota/iri/cache/impl/CacheImpl.java [hunk 1]
```diff
@@ -100,6 +100,9 @@ public int getSize() {
 
     @Override
     public void put(K key, V value) {
+        if (key == null || value == null) {
+            return;
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/cache/impl/CacheImpl.java]: Something is strange to me... if we look at `get()` we see
``` java
if (value == null && weakStore.containsKey(key)) {
            value = weakStore.get(key);
            put(key, value);
            weakStore.remove(key);
        }
```
This `put` failed meaning that that the condition returned true...
I guess what happened is that `weakStore.containsKey(key)` returned true -> GC kicked in -> failure at put

I am afraid that you are just masking the problem here... paving the way to more latent bugs
I would say that fixing `get()` as so would be better:

``` java
if (value == null) {
            value = weakStore.get(key);
            if (value != null) {
               put(key, value);
               weakStore.remove(key);
        }
```

Comment (GalRogozinski) [reply src/main/java/com/iota/iri/cache/impl/CacheImpl.java]: perhaps it is worth checking if at other places we forget that GC can kick in...

Comment (acha-bill) [reply src/main/java/com/iota/iri/cache/impl/CacheImpl.java]: Makes sense.

Comment (GalRogozinski) [reply src/main/java/com/iota/iri/cache/impl/CacheImpl.java]: > I am afraid that you are just masking the problem here
I would delete this change

Even though you are preventing, the error from happening you are just hiding the underlying cause
I would rather have an exception thrown in testing so we can do a proper fix when this happens

**Pull request merged**